### PR TITLE
Cohort self-selection and testing.

### PR DESCRIPTION
**Pull request by lucyq**
Merging to master cohort self selection with selenium tests.

### CODE REVIEW

**Comment by frederickjansen** [comment]
Just making sure I understand the history here. This is pacesetters with jiff, merged with cohort selection?

**Comment by lucyq** [comment]
@frederickjansen yes! With tests that I'm currently writing, which are based off of the selenium ones that have been merged to master.

File: server/production.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+module.exports = {
+  create: function (app) {
+    if (process.env.NODE_ENV !== 'production') { // Staging
+      // Do not force https, use a regular http server
+      return http.createServer(app);
+    }
+
+    // Production: use https
+    const LEX = require('greenlock-express');
```
Comment (frederickjansen) [inline server/production.js]: HTTPS broke at some point and needs to be either fixed (update dependency) or removed and tied to Nginx as a reverse proxy.

File: server/modules/modules.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
+/**
+ * Database modules
```
Comment (frederickjansen) [inline server/modules/modules.js]: Let's rename to models before we merge

File: server/modules/modules.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ */
+
+'use strict';
+
+const mongoose = require('mongoose');
+const Promise = require('bluebird');
```
Comment (frederickjansen) [inline server/modules/modules.js]: Mongoose has its own promise library. Do we need this?

File: server/modules/modulesWrappers.js [hunk 4]
```diff
@@ -0,0 +1,236 @@
+const Promise = require('bluebird');
```
Comment (frederickjansen) [inline server/modules/modulesWrappers.js]: Let's use native node promises?

**Comment by lucyq** [comment]
Commit closes #157

**Pull request merged**